src/test/java/api/loans/scenarios/ChangeDueDateAPITests.java

```diff
@@ -260,6 +263,25 @@ public void changeDueDateNoticeIsSentWhenPolicyIsDefined() {
       hasEmailNoticeProperties(steve.getId(), templateId, matchers)));
   }
 
+  @Test
+  public void dueDateChangedEventIsPublished() {
+    final DateTime newDueDate = dueDate.plus(Period.days(14));
+    changeDueDateFixture.changeDueDate(new ChangeDueDateRequestBuilder()
+      .forLoan(loan.getId())
+      .withDueDate(newDueDate));
+
+    Response response = loansClient.getById(loan.getId());
+    JsonObject updatedLoan = response.getJson();
+
+    List<JsonObject> publishedEvents = Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(FakePubSub::getPublishedEvents, hasSize(2));
+
+    JsonObject event = publishedEvents.get(1);
+
+    assertThat(event, isValidLoanDueDateChangedEvent(updatedLoan));
```

Comment:
Should this event also be published in other circumstances where the due date of the loan changes?

Reply:
I only know of two such circumstances - "due date change" and "recall request", the event is published in both cases. Are there others?

Reply:
> I only know of two such circumstances - "due date change" and "recall request", the event is published in both cases. Are there others?

A renewal will likely change the due date. That might be considered different and separate.

I can't remember if a request cancellation or move might also change the due date.

Reply:
I tried to track all of such places and as a result added publishing of "due date changed" event to:
- RenewalResource
- LoanCollectionResource.create
- RequestCollectionResource.replace
- RequestCollectionResource.move

Reply:
Have the permissions for publishing a message to pub-sub been added to each endpoint where this code is executed?